File: projects/10/java/WalkingBird.java
/* WalkingBird.java provides an abstract WalkingBird subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Josiah Ryan
 * Date:         04/14/21
 ******************************************************/

public abstract class WalkingBird extends Bird
{
    
  /* default constructor
  * PostCond: myName == "".
  */
  WalkingBird() {
    super();
  }

  /* explicit constructor
  * Receive: name, a String 
  * PostCond: myName == name.
  */
  WalkingBird(String name) {
    super(name);
 }

  /* movement() returns the kind of movement the bird does
   * Return "walked past"
   */ 
  public String movement() {
    return "walked past";
  }
}

File: projects/10/ruby/WalkingBird.rb
# WalkingBird.rb | Defines a Bird subclass 
#   to be extended by specific walking bird sub-classes.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Josiah Ryan
# Date:          04/15/21
####################################################

require './Bird.rb'

class WalkingBird < Bird

    # this extends the movement() method in the Bird super class
    # returns "walked by"
    def movement()
        return "walked by"
    end
  
  end